op-program/client/l2/engineapi/block_processor.go

```diff
@@ -51,6 +51,15 @@ func NewBlockProcessorFromPayloadAttributes(provider BlockDataProvider, parent c
 		Nonce:            types.EncodeNonce(0),
 		ParentBeaconRoot: attrs.ParentBeaconBlockRoot,
 	}
+	if attrs.EIP1559Params != nil {
+		// Do we need to check if holocene is active?
```

Comment:
Good question. We could acquire the holocene timestamp via `provider.Config().HoloceneTime`.

Reply:
cc @ajsutton. The geth team looks at the existence of fields in the header to know if a hardfork is active, we may be able to do the same thing here

Reply:
@tynes yes that was what I was thinking, just wasn't entirely sure, hence the comment there.

Reply:
I think it's fine as is but we should double check with more people

Reply:
As long as there's a header validation performed to ensure the field is always set for holocene (which I expect we need anyway) then yeah checking for the presence of the field makes a lot of sense (and fits my personal preference for letting the data structure do the talking instead of needing config everywhere)


Hmmm, I'm not seeing anything that does that validation in this PR. It may exist in the op-geth engine implementation but then I suspect it should have been add to `l2_engine_api.go`.

Reply:
So maybe we want that validation to happen inside of `verifyPayloadAttributes` in `l2_engine_api.go`?

Reply:
The validation of EIP1559Params and ExtraData is done in op-geth (execution API will return INVALID if they are set inappropriately).

Reply:
Just reopening this one to confirm. Could you point me to the geth code that does the validation please? op-program needs to implement an in-process version of the engine API so some of the checks have to be duplicated there, but we don't seem to have added this to `l2_engine_api.go` which is what op-program uses.

It's probably not a big deal because op-program also embeds op-node so something would be badly wrong if it set the params incorrectly but it's still generally good to ensure we have a consistent set of checks.

Reply:
Sorry @sebastianst pointed out in Discord I misunderstood the question.  I don't think there is any validation in there is any validation going on in the op-program engine. Not too familiar with how all this works but let me look into it.

Reply:
OK I see the engine API is basically duplicated in l2_engine_api.go and have added the same validation checks there. Let me know if more is required.

Reply:
Yep that covers it.  Thanks.

Reply:
thanks for confirming, resolving and merging